src/Analyzers/MSTest.Analyzers/ClassInitializeShouldBeValidAnalyzer.cs

```diff
@@ -62,7 +63,20 @@ public override void Initialize(AnalysisContext context)
         INamedTypeSymbol? valueTaskSymbol, INamedTypeSymbol? testContextSymbol, bool canDiscoverInternals)
     {
         var methodSymbol = (IMethodSymbol)context.Symbol;
-        if (!methodSymbol.IsClassInitializeMethod(classInitializeAttributeSymbol))
+        var namedTypeSymbol = context.Symbol.ContainingType;
+
+        bool isClassInitAttrExist = false;
+        bool isInheritanceModeSet = false;
+        foreach (AttributeData attr in methodSymbol.GetAttributes())
+        {
+            if (SymbolEqualityComparer.Default.Equals(attr.AttributeClass, classInitializeAttributeSymbol))
+            {
+                isClassInitAttrExist = true;
+                isInheritanceModeSet = attr.ConstructorArguments.Length == 1 ? true : false;
```

Comment:
what if `[ClassInitialize(InheritanceBehavior.None)]`

Reply:
Yeah, the current implementation is quite brittle if the attribute evolves.